src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1018,7 +1018,9 @@ namespace nupic {
     for (UInt i = 0; i < numColumns; i++) {
       Real perm[8];
       sp.getPermanence(i, perm);
-      NTA_CHECK(check_vector_eq(truePermArr[i], perm, numInputs));
+      print_vec(truePermArr[i], 8);
+      print_vec(perm, 8);
+      NTA_CHECK(check_vector_eq(truePermArr[i], perm, numInputs)) << "Failing: " << truePermArr[i] << " vs. " << perm;
```

Comment:
Nah, I wanted to know why you think the test is failing in context of this PR, but seems @oxtopus nailed it down to an error in calling `pow()` (which will be removed)